libafl_frida/src/asan/asan_rt.rs

```diff
@@ -514,18 +488,13 @@ impl AsanRuntime {
                         //is this necessary? The stalked return address will always be the real return address
                      //   let real_address = this.real_address_for_stalked(invocation.return_addr());
                         let original = [<$name:snake:upper _PTR>].get().unwrap();
-                        if this.hooks_enabled {
-                            let previous_hook_state = this.hooks_enabled;
-                            this.hooks_enabled = false;
+                        if !ASAN_IN_HOOK.get() && this.hooks_enabled { //not in hook and hooks are enabled
```

Comment:
Why did you change `this.hooks_enabled` to `ASAN_IN_HOOK.get()`?

Reply:
There is an issue in the test with`operator new`. `operator new` runs with the second matching case for hook which sets `ASAN_IN_HOOK` to false. However, `hooks_enabled` is still true, so when mmap runs in `alloc`, it his `hook_mmap` which unpoisons the entire allocated region causing the test to fail.